pkg/cli/gptscript.go

```diff
@@ -338,17 +338,32 @@ func (r *GPTScript) Run(cmd *cobra.Command, args []string) (retErr error) {
 				return fmt.Errorf("chat UI only supports files, cannot read from stdin")
 			}
 
-			absPathToScript, err := filepath.Abs(args[1])
-			if err != nil {
-				return fmt.Errorf("cannot determine absolute path to script %s: %v", args[1], err)
+			file := args[1]
+
+			// If the file is external, then set the SCRIPTS_PATH to the current working directory. Otherwise,
+			// set it to the directory of the script and set the file to the base.
+			external := strings.HasPrefix(file, "http://") || strings.HasPrefix(file, "https://") || strings.HasPrefix(file, "github.com")
+			if !external {
+				absPathToScript, err := filepath.Abs(args[1])
```

Comment:
Since we have `file` above, can we use it here? I imagine someone changing the `file` variable and being confused with that change not being reflected here. Same as below on line 352.